### PR TITLE
Add pypy to CI test matrix

### PR DESCRIPTION
**Pull request by hukkin**

### CODE REVIEW

**Comment by hukkin** [comment]
Shoutout to @graingert for making me revisit this PR. :wink:

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9, 3.10-dev]
+        python-version: [pypy-3.6, 3.6, 3.7, 3.8, 3.9, 3.10-dev]
```
Comment (graingert) [inline .github/workflows/tests.yaml:36]:
```suggestion
        python-version: [pypy-3.6, 3.6, 3.7, pypy-3.7, 3.8, 3.9, 3.10-dev]
```

**Review by graingert — CHANGES_REQUESTED** [review]
.

**Comment by graingert** [comment]
@hukkin you should add an "allgood" job so you don't need to update the required builds: https://github.community/t/is-it-possible-to-require-all-github-actions-tasks-to-pass-without-enumerating-them/117957/4

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (hukkin) [reply .github/workflows/tests.yaml:36]: Thanks, do you know about GitHub CIs usage limits? If there's no risk of running out then this is great

Comment (graingert) [reply .github/workflows/tests.yaml:36]: you can call them up all q.q like and get more minutes

Comment (hukkin) [reply .github/workflows/tests.yaml:36]: Haha good to know

**Comment by hukkin** [comment]
> you should add an "allgood" job

You my friend are an absolute genius! I didn't know I needed this but I do.

Also thanks for all the effort on making Python ecosystem TOML v1 friendly

**Review by graingert — APPROVED** [review]

**Pull request merged**